### PR TITLE
Taegis: Add Asset and Alert Update Commands

### PR DESCRIPTION
**Pull request by reedium**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Adds support for the following commands related to assets and alerts:
* taegis-fetch-assets
* taegis-fetch-endpoint
* taegis-isolate-asset
* taegis-update-alert-status

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 1]
```diff
@@ -377,6 +396,84 @@ def fetch_alerts_command(client: Client, env: str, args=None):
     return results
 
 
+def fetch_assets_command(client: Client, env: str, args=None):
+    page = int(args.get("page", 0))
+    page_size = int(args.get("page_size", 10))
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:401]:
```suggestion
    page = arg_to_number(args.get("page", 0))
    page_size = arg_to_number(args.get("page_size", 10))
```
Almost the same, just in the arg_to_number command we handle some edge cases :)

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 2]
```diff
@@ -377,6 +396,84 @@ def fetch_alerts_command(client: Client, env: str, args=None):
...
+            "limit": page_size,
+            "offset": page_size * page,
+        }
+    }
+
+    # Loop over allowed search fields and add valid search options to the query variables
+    for field in ASSET_SEARCH_FIELDS:
+        if not args.get(field):
+            continue
+        variables["input"][field] = args.get(field).strip()
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:415]:
```suggestion
        if args.get(field):
            variables["input"][field] = args.get(field).strip()
```

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py [hunk 3]
```diff
@@ -883,6 +1030,104 @@ def fetch_users_command(client: Client, env: str, args=None):
...
+
+    variables = {
+        "alert_ids": args.get("ids"),
+        "reason": args.get("reason", ""),
+        "resolution_status": args.get("status"),
+    }
+
+    if type(variables["alert_ids"]) == str:
+        variables["alert_ids"] = variables["alert_ids"].split(",")  # alerts://id1,alerts://id2
+    variables["alert_ids"] = [x.strip() for x in variables["alert_ids"]]  # Ensure no whitespace
```
Comment (yucohen) [inline Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py:1093]: is `variables["alert_ids"]` should always be a list? even if we have only one id?
If so, you can just use argToList:
`"alert_ids": argToList(args.get("ids"))`
In any case, use a variable for `args.get("ids")`, as now, you change the value of a dict in runtime, and it is not a best-practice (and can cause issues)

**Review by yucohen — CHANGES_REQUESTED** [review]
@reedium Thanks for your contribution! great work!
Please see my comments :)

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Development container for Content. Poetry hash 97b4446a28329c24352fcb41ca8570403da0ae3f7434121ab73b1ff83fdec395
+// Development container for Content. Poetry hash d0f1eeb8380e9a8d99a91ba51c6310129fa3577c918219946ffc47450b97ca85
```
Comment (yucohen) [inline .devcontainer/devcontainer.json:1]: please revert this change

Comment (reedium) [reply .devcontainer/devcontainer.json]: Sorry about that. Not sure how that commit ended up in my branch. Reverted in 93a10b7

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py (hunk 3 above)
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py]: I had no idea about argToList. That definitely simplifies things. Thank you for the heads up.

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py (hunk 2 above)
Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py]: Sounds good. Updated.

File: Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py (hunk 1 above)
Comment (yucohen) [reply Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py]: try this:
    page = arg_to_number(args.get("page")) or 0
    page_size = arg_to_number(args.get("page_size")) or 10

Comment (reedium) [reply Packs/SecureWorks/Integrations/TaegisXDR/TaegisXDR.py]: Gotcha. That took care of it. Thinks for the help.

**Comment by yucohen** [comment]
@reedium Great work with the fixes, merging :)

**Review by yucohen — APPROVED** [review]
